internal/resources/aws/cloudfront_distribution.go

```diff
@@ -0,0 +1,659 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/shopspring/decimal"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/infracost/infracost/internal/usage"
+)
+
+type CloudfrontDistribution struct {
+	Address string
+	Region  string
+
+	IsOriginShieldEnabled     bool
+	IsSSLSupportMethodVIP     bool
+	HasLoggingConfigBucket    bool
+	HasFieldLevelEncryptionID bool
+	OriginShieldRegion        *string
```

Comment:
```suggestion
	OriginShieldRegion        string
```
I think this could just be a string and we can check later if it is non-empty.

Reply:
Updated 👍